### ISSUE TITLE
When user submits answer, goes to next question, and comes back, the state doesn't remain

### ISSUE THREAD

**Issue by austintackaberry** [austintackaberry/ydkjs-exercises#108]
If the user answers a questions (correctly or incorrectly), the state of that page should not be any different than it was when they left.

![awesomescreenshot-2018-07-02t00-04-06-824z](https://user-images.githubusercontent.com/29493001/42140218-fb1142bc-7d50-11e8-97af-71e9b3611bef.gif)

**Comment by radotreyes** [comment]
I'm curious as to what the best solution to this is. I actually want to say that we should allow users to re-answer questions if they want to (say they want to review or something), but I agree that there should be some indicator that the user has interacted with that question before, like a banner indicating as such? What do you think?

**Comment by austintackaberry** [comment]
I definitely think that users should be able to re-answer as many times as they like. But I think users would want to remember what they answered. Especially when they go to the next question and then go right back.

Maybe we could have a "reset" option that wipes away the state. This option would be for users who are using it like a quiz.

**Comment by austintackaberry** [comment]
I don't think we will have to use a Map. The `questionId` is a string, so a plain old object will do. Since we lose the question state every time we move to a new question, I think we would have to use React Context API for this.

We could save this info in `bookScores` -> (https://github.com/austintackaberry/ydkjs-exercises/blob/master/src/score-context.js)

Or we could make a whole new object that only has `questionId` as the key and whatever state we want to keep as the value.

My personal opinion is that we should save it in `bookScores` since we are already saving some question state in there (correct, incorrect, or unanswered)
